src/test/java/hwjdi/MetalsAndColorsJdiTest.java

```diff
@@ -0,0 +1,26 @@
+package hwjdi;
+
+import hwjdi.base.BaseClass;
+import hwjdi.entities.MetalsAndColors;
+import org.testng.annotations.Test;
+import static hwjdi.entities.User.*;
+import static org.testng.Assert.assertEquals;
+
+public class MetalsAndColorsJdiTest extends BaseClass {
+
+    @Test(dataProviderClass = MetalsAndColorsDataProvider.class, dataProvider = "testData")
+    public void jdiMetalsAndColorsTest(MetalsAndColors metalsAndColors) {
+
+        JdiSite.openHomePage();
+        JdiSite.jdiHomePage.login(ROMAN);
+        JdiSite.jdiHomePage.userName.is().text(ROMAN.getFullName());
+
+        JdiSite.openMetalsColorsPage();
+        JdiSite.metalsAndColorsPage.setWebElements(metalsAndColors);
+
+        assertEquals(JdiSite.metalsAndColorsPage.getResults(),
+                metalsAndColors.toString().replaceAll("\\[|\\]", ""));
```

Comment:
This replacement should be encapsulated somewhere else rather than test method.

Reply:
fixed